src/XMonad/Operations.hs

```diff
@@ -496,6 +497,31 @@ initColor dpy c = C.handle (\(C.SomeException _) -> return Nothing) $
     (Just . color_pixel . fst) <$> allocNamedColor dpy colormap c
     where colormap = defaultColormap dpy (defaultScreen dpy)
 
+-- | Use a 'Monad' function to modify the workspaces in 'XState', starting with the current workspace.
+--
+-- This is effectively the same as 'runOnWorkspaces' but with a different order of operations and more general signature.
+runOnWorkspaces' ::
+    (MonadState XState m)=>
+    (WindowSpace -> m WindowSpace) -> m ()
+runOnWorkspaces' = windowsetM_ . workspacesA
+  where
+    windowsetM_ f = do
+        ws' <- f =<< gets windowset
+        modify $ \ s -> s{ windowset = ws' }
+
+-- | Traverse an 'Applicative' function over the workspaces, starting with the current.
+workspacesA ::
+    (Applicative m)=>
+    (W.Workspace i l a -> m (W.Workspace i' l' a)) ->
+    W.StackSet i l a sid sd -> m (W.StackSet i' l' a sid sd)
```

Comment:
This formatting (ditto `runOnWorkspaces'`) looks kind of funny

Reply:
Feel free to change it if you have a better formatting (do you need / can I give permission?). For long signatures like this I do my best to wrap them in semantic units that fit in a Tall layout on my little laptop screen. With an `Applicative` constraint, give a function from a workspace to an action on a workspace to get a function from a stackset to an action on a stackset.

Reply:
Once you convert a branch on your repo into a pull request, any xmonad maintainer has access to the PR. Of course, you're asking for trouble with force pushes in that case.

Reply:
You can tick the "allow access from maintainers" checkbox somewhere in the PR, but I think it is on by default.  I think the grouping is quite understandable; I was just thrown off a bit due to the trailing arrow syntax, as opposed to something like 

``` haskell
workspacesA
    :: Applicative m
    => (W.Workspace i l a -> m (W.Workspace i' l' a))
    -> W.StackSet i l a sid sd -> m (W.StackSet i' l' a sid sd)
```

Reply:
Ironically the prefix form is confusing to me because the type symbol `::` is on the constraint, the constraint arrow `=>` is on a parameter, and a parameter arrow `->` is on the result. But that does seem to be the norm for XMonad, so I'll switch it over.